liquigraph-core/src/main/java/org/liquigraph/core/io/lock/LockableConnection.java

```diff
@@ -42,28 +46,42 @@
  *
  * A shutdown hook is executed to remove the lock node if and
  * only if the connection has not been properly closed.
+ *
+ * Please note that any {@link Connection} passed to this decorator
+ * will be set to auto-commit: false`. The auto-commit property
+ * is restored on close. The latter is done in order to minimize
+ * side effects when connections are recycled by a connection pool,
+ * for instance.
  */
 public final class LockableConnection implements Connection {
     private final Connection delegate;
     private final LiquigraphLock lock;
+    private final boolean previousAutoCommit;
 
-    private LockableConnection(Connection delegate, LiquigraphLock lock) {
+    private LockableConnection(Connection delegate, boolean previousAutoCommit, LiquigraphLock lock) {
         this.delegate = delegate;
+        this.previousAutoCommit = previousAutoCommit;
         this.lock = lock;
     }
 
     public static LockableConnection acquire(Connection delegate, LiquigraphLock lock) {
-        LockableConnection connection = new LockableConnection(delegate, lock);
+        LockableConnection connection = null;
         try {
+            boolean previousAutoCommit = delegate.getAutoCommit();
+            delegate.setAutoCommit(false);
+
+            connection = new LockableConnection(delegate, previousAutoCommit, lock);
             lock.acquire(connection);
             return connection;
-        } catch (RuntimeException e) {
+        } catch (SQLException | RuntimeException e) {
             try {
-                connection.close();
+                if (connection != null) {
```

Comment:
I've realized there's another failure mode, which needs to be dealt with here: `delegate.setAutoCommit(false)` can throw a `SQLException`, `connection` will be null, but `delegate` should be closed nonetheless.

That can be done here or in the caller, i.e. `GraphJdbcConnector::connect`, but in the caller it would need to check if it's not been closed yet (which could happen if `lock.acquire(connection)` fails). I think it's cleaner here:
```java
if (connection == null) {
    delegate.close();
} else {
    connection.close();
}
```